torch/csrc/jit/frontend/sugared_value.cpp

```diff
@@ -157,6 +157,17 @@ std::shared_ptr<SugaredValue> SimpleValue::attr(
       auto n = g.insertNode(g.createGetAttr(value_, field));
       return std::make_shared<SimpleValue>(n->output());
     }
+    // Check and see if it's a getter attribute.
+    std::string getter_name = field + "_getter";
+    auto* getter = classType->findMethod(getter_name);
+    if (getter) {
+      auto& g = *m.graph();
```

Comment:
Can you refactor this to `MethodValue(value_,  getter_name)->call(...)` this reuses the error checking for recursive calls and such